routing/localchans/manager.go

```diff
@@ -76,20 +76,34 @@ func (r *Manager) UpdatePolicy(newSchema routing.ChannelPolicy,
 			return nil
 		}
 
+		// Add edge to a list. We can't yet apply the new policy,
+		// because the updateEdge method will open a new db transaction.
+		edges = append(edges, discovery.EdgeWithInfo{
```

Comment:
I think `edges` should be initialized on first access. We can't be sure if `ForAllOutgoingChannels` is ever retrying. Even though all of this is going to be ram cached soon.